src/collection-view.js

```diff
@@ -169,19 +120,21 @@ var CollectionView = Backbone.View.extend({
     return this;
   },
 
-  // An efficient rendering used for filtering. Instead of modifying
-  // the whole DOM for the collection view, we are only adding or
-  // removing the related childrenViews.
-  setFilter: function(filter, options = {}) {
-    var viewCanBeRendered = this._isRendered && !this._isDestroyed;
+  // An efficient rendering used for filtering. Instead of modifying the whole DOM for the
+  // collection view, we are only adding or removing the related childrenViews.
+  setFilter(filter, {preventRender} = {}) {
+    const viewCanBeRendered = this._isRendered && !this._isDestroyed;
```

Comment:
Great observation.  `filter` should indeed be set no matter what.  I have improved the logic like so:

```js
    const canBeRendered = this._isRendered && !this._isDestroyed;
    const filterChanged = this.filter !== filter;
    const shouldRender = canBeRendered && filterChanged && !preventRender;
    // if shouldRender, render and set this.filter, otherwise set this.filter
```